### PR TITLE
Add more missing test scenarios and verifications

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Issue:** https://github.com/stolostron/backlog/issues/25059

### Changes:
- Implement several new tests senarios for queries and rbac.

image
image

### CODE REVIEW

File: tests/api/rbac.test.js [hunk 1]
```diff
@@ -50,64 +62,125 @@ describe(`[P2][Sev2][${squad}] Search API: Verify RBAC`, () => {
...
+    test('should validate RBAC configuration.', () => {
+      expectCli(`oc auth can-i list secret -n ${ns} --as=${user.fullName}`).toThrow()
+      expectCli(`oc auth can-i list pod -n ${ns} --as=${user.fullName}`).toThrow()
+      expectCli(`oc auth can-i list deployment -n ${ns} --as=${user.fullName}`).not.toThrow()
+    })
+
+    test(`should not get Secret`, () => ValidateSearchData({ user, kind: 'secret', namespace: ns }), validationTimeout)
+    test('should not get Pod', () => ValidateSearchData({ user, kind: 'configmap', namespace: ns }), validationTimeout)
+    test(
+      'should get Deploymt',
```
Comment (dislbenn) [inline tests/api/rbac.test.js:175]:
```suggestion
      'should get Deployment',
```

Deployment is spelled wrong.

**Comment by dislbenn** [comment]
@jlpadilla I also saw within the `ci/prow/deploy-and-run-e2e` test logs the following: 

```bash
Command "get-token" is deprecated and will be removed in the future version. Use oc create token instead.
```

We might need to keep an eye on this just in case it becomes an issue in the future.

File: tests/api/rbac.test.js (hunk 1 above)
Comment (jlpadilla) [reply tests/api/rbac.test.js:175]: This was on purpose trying to keep it in a single line, but it went over anyways. :-(

**Comment by jlpadilla** [comment]
@dislbenn Thanks! I guess this started after I updated the OC version. I'll switch to oc create token ASAP because this is what caused the canaries to fail on OCP 4.11

> @jlpadilla I also saw within the `ci/prow/deploy-and-run-e2e` test logs the following:
> 
> ```shell
> Command "get-token" is deprecated and will be removed in the future version. Use oc create token instead.
> ```
> 
> We might need to keep an eye on this just in case it becomes an issue in the future.

**Comment by jlpadilla** [comment]
/unhold

Will address @dislbenn 's comments in the next PR.

**Pull request merged**